odoo/api.py

```diff
@@ -611,6 +611,9 @@ def clear(self):
         """ Clear all record caches, and discard all fields to recompute.
             This may be useful when recovering from a failed ORM operation.
         """
+        # Since the company/companies lazy properties can contain data that is no longer valid,
+        # (e.g. company created during a test or a failed ORM operation), reset these lazy properties
+        lazy_property.reset_all(self.all)
```

Comment:
This is not correct: the lazy properties are not on `self.all`, but on all environments in `list(self.all)`.

Reply:
```suggestion
        for env in list(self.all):
            lazy_property.reset_all(env)
```